src/engine/MaxConstraint.cpp

```diff
@@ -403,204 +370,71 @@ unsigned MaxConstraint::getF() const
 
 bool MaxConstraint::satisfied() const
 {
-    if ( !( _assignment.exists( _f ) && _assignment.size() > 0 ) )
-        throw MarabouError( MarabouError::PARTICIPATING_VARIABLES_ABSENT );
+    DEBUG({
+            if ( !( _assignment.exists( _f ) ) )
+                throw MarabouError
+                    ( MarabouError::PARTICIPATING_VARIABLES_ABSENT,
+                      Stringf( "f(x%u) assignment missing.", _f ).ascii() );
+            for ( const auto &element : _elements )
+                if ( !( _assignment.exists( element ) ) )
+                    throw MarabouError
+                        ( MarabouError::PARTICIPATING_VARIABLES_ABSENT,
```

Comment:
In the Error name, absent implies the variable is missing, while in reality assignment is missing. So maybe replace ABSENT with MISSING_ASSIGNMENT?

Reply:
Renamed to participating_var_missing_assignment